packages/app/components/stackbox/Stackbox.tsx

```diff
@@ -327,13 +328,21 @@ export const Stackbox = () => {
             </div>
           </div>
         </div>
-        <Button
-          width="full"
-          onClick={openConfirmStack}
-          className={cx({ "animate-wiggle": showInsufficentBalanceError })}
-        >
-          {showInsufficentBalanceError ? "Insufficent Balance" : "Stack Now"}
-        </Button>
+        {address ? (
+          <Button
+            width="full"
+            onClick={openConfirmStack}
+            className={cx({ "animate-wiggle": showInsufficentBalanceError })}
+          >
+            {showInsufficentBalanceError ? "Insufficent Balance" : "Stack Now"}
+          </Button>
+        ) : (
+          <ConnectButton
+            // size="lg"
+            className="w-full"
+            text="Connect wallet to stack"
```

Comment:
Unwanted comment and maybe text should be: "Connect Wallet to Stack" or just "Connect Wallet".